akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -333,17 +342,16 @@ private[stream] final class GraphInterpreterShell(
    * to give each input buffer slot a chance to run through the whole pipeline
    * and back (for the demand).
    */
-  private val eventLimit = settings.maxInputBufferSize * (assembly.ins.length + assembly.outs.length)
-
+  val shellEventLimit = settings.maxInputBufferSize * (assembly.ins.length + assembly.outs.length)
```

Comment:
Just thought about this—I had the feeling that it should be “at least N” with N > 1, but then I couldn’t back this up with facts. Now I think I have found an example of why this is the upper limit that we can choose, and if this is correct then it should be added to the explaining comment:

* assume a composite Sink of one `expand` and one `fold`
* assume an infinitely fast source of data
* assume maxInputBufferSize == 1 (i.e. the user when the whole nine yards to get more fairness)
* if the event limit is greater than what we calculate here then there will always be `expand` activity because no data can enter “fast enough” from the outside